src/org/zaproxy/zap/extension/pscanrulesBeta/ServletParameterPollutionScanner.java

```diff
@@ -75,20 +73,10 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 			// check for 'target' param
 			
 			for (Element formElement : formElements) {
-				boolean actionFound = false;
-				Attributes atts = formElement.getAttributes();
-				Iterator<Attribute> iter = atts.iterator();
-				
-				while (iter.hasNext()) {
-					Attribute att = iter.next();
-					if (att.getName().equalsIgnoreCase("action") && att.getValue().length() > 0) {
-						// action tag present (and with a value), so should be ok
-						actionFound = true;
-					}
-				}
-				
-				if (!actionFound) {
-				    Alert alert = new Alert(getPluginId(), Alert.RISK_MEDIUM, Alert.CONFIDENCE_LOW, 
+				boolean actionMissingOrEmpty = StringUtils.isEmpty(formElement.getAttributeValue("action"));
+
+				if (actionMissingOrEmpty) {
```

Comment:
I find the variable improves readability, but that is personal preference. Let me know what you think.

And yeah the call can return null, for example when the element  is ``` <form> ``` with no action attribute.